specification/cognitiveservices/data-plane/Language/preview/2021-11-01-preview/textanalytics.json

```diff
@@ -0,0 +1,2106 @@
+                            "items": {
+                                "$ref": "#/definitions/EntityLinkingTask"
+                            }
+                        },
+                        "sentimentAnalysisTasks": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/SentimentAnalysisTask"
+                            }
+                        },
+                        "extractiveSummarizationTasks": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/ExtractiveSummarizationTask"
+                            }
+                        }
+                    },
+                    "type": "object"
+                }
+            },
+            "required": [
+                "tasks"
+            ],
+            "type": "object"
+        },
+        "Task": {
+            "type": "object",
+            "description": "Base task object.",
+            "properties": {
+                "taskName": {
+                    "type": "string"
+                }
+            }
+        },
+        "TaskParameters": {
+            "type": "object",
+            "description": "Base parameters object for a text analysis task.",
+            "properties": {
+                "loggingOptOut": {
+                    "type": "boolean",
+                    "default": false
+                }
+            }
+        },
+        "PreBuiltTaskParameters": {
+            "type": "object",
+            "description": "Parameters object for a text analysis task using pre-built models.",
+            "properties": {
+                "modelVersion": {
+                    "type": "string",
+                    "default": "latest"
+                }
+            }
+        },
+        "PreBuiltResult": {
+            "properties": {
+                "errors": {
+                    "type": "array",
+                    "description": "Errors by record id.",
+                    "items": {
+                        "$ref": "#/definitions/RecordError"
+                    }
+                },
+                "statistics": {
+                    "$ref": "#/definitions/RequestStatistics"
+                },
+                "modelVersion": {
+                    "type": "string",
+                    "description": "This field indicates which model is used for scoring."
+                }
+            },
+            "required": [
+                "errors",
+                "modelVersion"
+            ]
+        },
+        "CustomTaskParameters": {
+            "type": "object",
+            "description": "Parameters object for a text analysis task using custom models.",
+            "properties": {
+                "project-name": {
+                    "type": "string"
+                },
+                "deployment-name": {
+                    "type": "string"
```

Comment:
I think we need describe the input limits for all string inputs in the swagger.

Reply:
I think this might take some thought, since a lot of these strings depend on different things.  It's possible I'm over thinking things a bit, but just want to make sure that we set the values appropriately now so that if they change later we won't break clients.

Reply:
@johanste , @mikekistler , do either of you have any guidance for choosing limits for string lengths?

Reply:
I think there is some ambiguity in terms of how to set string input limits on some objects, since some objects will be reused in multiple scenarios which may have different input limits.

I'm going to resolve this for now since I didn't get guidance from the API board, but let's have a deeper discussion about it at some point in the future if it's really important to have this.